### PR TITLE
Add debug log for check-in miss

### PR DESCRIPTION
**Pull request by andrzej-stencel**
## What does this PR do?

I want to log the time details to debug flaky tests like https://github.com/elastic/elastic-agent/issues/4203.

I'm not sure if this is the right function to call, happy to learn from others.

## Why is it important?

When a flaky tests like https://github.com/elastic/elastic-agent/issues/4203 fails, it helps to know as many details about the environment as possible.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

### CODE REVIEW

File: pkg/component/runtime/command.go [hunk 1]
```diff
@@ -238,6 +238,7 @@ func (c *commandRuntime) Run(ctx context.Context, comm Communicator) error {
 						c.missedCheckins = 0
 					} else {
 						c.missedCheckins++
+						c.logStd.Write([]byte(fmt.Sprintf("Last check-in was: %s, now is: %s. The diff %s is higher than allowed %s.", c.lastCheckin.Format(time.RFC3339Nano), now.Format(time.RFC3339Nano), now.Sub(c.lastCheckin), checkinPeriod)))
```
[the remark this answers is not in the document]
Comment (andrzej-stencel) [reply pkg/component/runtime/command.go:241]: That would make sense and I would love to use a Debug function instead, but `logStd` is for some reason not a `logp.Logger`, but a [logWriter](https://github.com/elastic/elastic-agent/blob/5dc0ddb6529b0a0ec16ac1672e72e71714aa1e98/pkg/component/runtime/log_writer.go#L35): [https://github.com/elastic/elastic-agent/blob/main/pkg/component/runtime/command.go#L70](https://github.com/elastic/elastic-agent/blob/5dc0ddb6529b0a0ec16ac1672e72e71714aa1e98/pkg/component/runtime/command.go#L70).

[the remark this answers is not in the document]
Comment (leehinman) [reply pkg/component/runtime/command.go:241]: > Am I missing something? How should I go about this?

Sorry, you weren't missing anything.  I was.

[the remark this answers is not in the document]
Comment (andrzej-stencel) [reply pkg/component/runtime/command.go:241]: I'm still not quite sure if calling `c.logStd.Write` here is the right or the best way to log the information we need. Any thoughts or suggestions are welcome.

[the remark this answers is not in the document]
Comment (pchila) [reply pkg/component/runtime/command.go]: `createLogWriter` takes a `logp.Logger` as argument originally passed in `newCommandRuntime`, if I read the code correctly we use those writers to stream the command stderr and stdout in the agent logs with the correct context.
I would hold a reference to this base log in the commandRuntime struct and use it for logging happening "on the agent side" (for example the missed checkins and other stuff that happens on agent side)

[the remark this answers is not in the document]
Comment (andrzej-stencel) [reply pkg/component/runtime/command.go]: Thank you @pchila! I like your suggestion a lot. I've applied it here: https://github.com/elastic/elastic-agent/pull/5348/commits/4c708a4504c25cb41eb5eb886d54626ca29b7a94. Let me know your thoughts.

**Review by pchila — APPROVED** [review]
LGTM

**Pull request merged**